pingidentity/pingctl
# PDI-1632: Add Testing Framework to PingCTL

- Update Exportable Resource ImportBlock to have Equals() and String() functions for testing.
- Rename resources.NotificationSettingsTemplateContent to resources.NotificationTemplateContent to fix terraform plan error
- Add resource exportAll() testing to pingone-platform resources
- Update pingone_key.go to use key type in resource name to avoid resource name duplication
- Full overhaul of resources.NotificationTemplateContent logic to export template content correctly
- Update PingoneRoleAssignmentUserResource to include the Role Name and Scope in the resource name for clarity. Switch to using username instead of given name to export all valid user roleAssignments
- Remove WSFED applications from export logic across all resources
- Update testutils helper functions to run once when initializing the API client and grabing the environment ID from env.
- Move common importblock validation to testutils helper function, and update logic to check for any duplicated resource names or ids
